src/main/java/picard/arrays/illumina/InfiniumDataFile.java

```diff
@@ -117,6 +118,19 @@ private int unsignedShortToInt(final byte[] bytes) {
         return integer;
     }
 
+    /**
+     * Utility method to convert a short to a byte array
+     *
+     * @param value The short value to convert to byte array
+     * @return The byte array containing the short in little endian format.
+     */
+    private static byte[] shortToByteArray(final short value) {
```

Comment:
Am surprised we don't have this already somewhere else. I think `ByteBuffer` can do conversions between byte arrays and java primitive types.

Reply:
FYI this is how to do this using `ByteBuffer`. IMO using the Java builtin would be better, but am not sure if would make the code slower.
```
    private static byte[] shortToByteArray(final short value) {
        ByteBuffer buffer = ByteBuffer.allocate(2).order(ByteOrder.LITTLE_ENDIAN);
        return buffer.putShort(value).array();
    }
```
It can convert from byte array as well:
```
    private static short byteArrayToShort(byte[] bytes) {
        ByteBuffer buffer = ByteBuffer.wrap(bytes).order(ByteOrder.LITTLE_ENDIAN);
        return buffer.getShort();
    }
```

Reply:
Okay - please check